docs/getting-started/concepts/feature-retrieval.md

```diff
@@ -1,14 +1,152 @@
 # Feature retrieval
 
-## Dataset
+## Overview
 
-A dataset is a collection of rows that is produced by a historical retrieval from Feast in order to train a model. A dataset is produced by a join from one or more feature views onto an entity dataframe. Therefore, a dataset may consist of features from multiple feature views.
+Generally, Feast supports several patterns of feature retrieval:
 
-**Dataset vs Feature View:** Feature views contain the schema of data and a reference to where data can be found \(through its data source\). Datasets are the actual data manifestation of querying those data sources.
+1. Training data generation (via `feature_store.get_historical_features(...)`)
+2. Offline feature retrieval for batch predictions (for example predicting user churn for all users on a daily basis)
```

Comment:
think it might be helpful to also list how to do this, similar to the (via function) syntax since it's also using get_historical_retrieval.